packages/material-ui/src/Paper/Paper.test.js

```diff
@@ -83,20 +82,16 @@ describe('<Paper />', () => {
     expect(getByTestId('root')).to.have.class('custom-elevation');
   });
 
-  describe('warnings', () => {
-    beforeEach(() => {
-      PropTypes.resetWarningCache();
-    });
-
-    it('warns if the given `elevation` is not implemented in the theme', () => {
-      expect(() => {
-        PropTypes.checkPropTypes(
-          Paper.Naked.propTypes,
-          { classes: { elevation24: 'elevation-24', elevation26: 'elevation-26' }, elevation: 25 },
-          'prop',
-          'MockedPaper',
-        );
-      }).toErrorDev('Material-UI: This elevation `25` is not implemented.');
-    });
+  it('warns if the given `elevation` is not implemented in the theme', () => {
+    const theme = createMuiTheme();
+    expect(() => {
+      render(
+        <ThemeProvider theme={theme}>
+          <Paper elevation={26} />
```

Comment:
elevation doesn't equals 25 because of this test:

https://github.com/mui-org/material-ui/blob/5bd54bce805b6186203a5ff6ab6a24df64f680a4/packages/material-ui/src/Paper/Paper.test.js#L73-L83

If there is any way to reset the theme, I will do that in `beforeEach` function.

Reply:
Don't add the `ThemeProvider` since it wasn't added in the original test.

Reply:
> elevation can't equal 25 because of this test:

In the future please explain what the issue is not just point to existing code. It wasn't obvious to me where and what the issue was. The specific issue is only at line 75: `theme.shadows.push`. I wasn't aware that this is still the reference to our default shadows.

I think we should change that though I'm filing this in another PR since it might affect other properties as well. Mutating the return value from `createMuiTheme` should only affect the return value and nothing more IMO.

Reply:
I'm sorry. I do more detailed and clearly explanation in the future. 

>Mutating the return value from createMuiTheme should only affect the return value and nothing more IMO.
I think so, too.

Reply:
> I'm sorry. 

All good. It's easily forgotten when digging into an issue that others don't build up all the context you have at the same time.